### PR TITLE
fix(#2738): multiple hook sidecars on the same hook point

### LINKED ISSUE #2738: Multiple hook sidecars cause virt-launcher crash
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

When virt-launcher has multiple hook sidecars (on the same `HookPoint`), virt-launcher crash with stacktrace:
```
{"component":"virt-launcher","level":"info","msg":"Collected all requested hook sidecar sockets","pos":"manager.go:68","timestamp":"2019-09-24T12:06:47.422796Z"}
panic: runtime error: index out of range

goroutine 1 [running]:
kubevirt.io/kubevirt/pkg/hooks.sortCallbacksPerHookPoint.func1(0x1, 0x0, 0x15bdc80)
	/root/go/src/kubevirt.io/kubevirt/pkg/hooks/manager.go:159 +0x117
sort.insertionSort_func(0xc00069fba8, 0xc0006dc480, 0x0, 0x2)
	/gimme/.gimme/versions/go1.11.2.linux.amd64/src/sort/zfuncversion.go:12 +0x93
sort.quickSort_func(0xc00069fba8, 0xc0006dc480, 0x0, 0x2, 0x4)
	/gimme/.gimme/versions/go1.11.2.linux.amd64/src/sort/zfuncversion.go:158 +0x1fb
sort.Slice(0x1221380, 0xc0006dc460, 0xc00069fba8)
	/gimme/.gimme/versions/go1.11.2.linux.amd64/src/sort/slice.go:21 +0x12f
kubevirt.io/kubevirt/pkg/hooks.sortCallbacksPerHookPoint(0xc0002dac60)
	/root/go/src/kubevirt.io/kubevirt/pkg/hooks/manager.go:158 +0xfd
kubevirt.io/kubevirt/pkg/hooks.(*Manager).Collect(0xc0002b0040, 0x2, 0x45d964b800, 0x1, 0xc0000b8638)
	/root/go/src/kubevirt.io/kubevirt/pkg/hooks/manager.go:70 +0xbb
main.main()
	/root/go/src/kubevirt.io/kubevirt/cmd/virt-launcher/virt-launcher.go:333 +0x5ee
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
```
func TestSort() {
	callbacksPerHookPoint := make(map[string][]*callBackClient)
	callbacksPerHookPoint["OnDefineDomain"] = append(callbacksPerHookPoint["OnDefineDomain"], &callBackClient{
		SocketPath: "/tmp/hook1.sock",
		Version:    "v1alpha1",
		subsribedHookPoints: []*hooksInfo.HookPoint{
			&hooksInfo.HookPoint{
				Name:     "OnDefineDomain",
				Priority: 0,
			},
		},
	})
	callbacksPerHookPoint["OnDefineDomain"] = append(callbacksPerHookPoint["OnDefineDomain"], &callBackClient{
		SocketPath: "/tmp/hook2.sock",
		Version:    "v1alpha1",
		subsribedHookPoints: []*hooksInfo.HookPoint{
			&hooksInfo.HookPoint{
				Name:     "OnDefineDomain",
				Priority: 0,
			},
		},
	})
	sortCallbacksPerHookPoint(callbacksPerHookPoint)
}
```

`sortCallbacksPerHookPoint` function is from https://github.com/kubevirt/kubevirt/blob/bc2d9deda24f4ed851539c87b0b3ede7a3ffa034/pkg/hooks/manager.go#L163-L175
maybe some mistake in this function, I think the correct one should looks like:

```
func sortCallbacksPerHookPoint(callbacksPerHookPoint map[string][]*callBackClient) {
	for _, callbacks := range callbacksPerHookPoint {
		for _, callback := range callbacks {
			sort.Slice(callback.subsribedHookPoints, func(i, j int) bool {
				if callback.subsribedHookPoints[i].Priority == callback.subsribedHookPoints[j].Priority {
					return strings.Compare(callback.subsribedHookPoints[i].Name, callback.subsribedHookPoints[j].Name) < 0
				} else {
					return callback.subsribedHookPoints[i].Priority > callback.subsribedHookPoints[j].Priority
				}
			})
		}
	}
}
```

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
```
Client Version: version.Info{GitVersion:"v0.14.0", GitCommit:"99129b309dae399061d38924744ded0ec0ea72c9", GitTreeState:"clean", BuildDate:"2019-02-04T13:49:57Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{GitVersion:"v0.14.0", GitCommit:"99129b309dae399061d38924744ded0ec0ea72c9", GitTreeState:"clean", BuildDate:"2019-02-04T13:49:57Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T15:22:13Z", GoVersion:"go1.9.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:00:57Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): `Ubuntu 18.04.2 LTS`
- Kernel (e.g. `uname -a`):`4.15.0-45-generic`
- Install tool
- Others:

### PR DESCRIPTION
**Pull request by lxs137**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** :
Fixes #2738

**Special notes for your reviewer**:
* fix mistake in `sortCallbacksPerHookPoint` function
* make `hooks.manager.CallbacksPerHookPoint` field public for testing.

**Release note**:

```release-note
Fix panic when more than one sidecar containers are added to a VMI
```

### CODE REVIEW

**Comment by rmohr** [comment]
@lxs137 could you run `make` and commit the changes? Travis thinks that the formatting is not correct: https://travis-ci.org/kubevirt/kubevirt/builds/589291178#L1434

**Comment by rmohr** [comment]
@lxs137 thanks a lot. Could you also add a release note in the PR description?

**Comment by rmohr** [comment]
Ha, sorry seems like you also have to run `make generate`: https://travis-ci.org/kubevirt/kubevirt/builds/589330269#L1199.

But then we are through with it.

File: pkg/hooks/manager_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+			manager := hooks.GetManager()
+			err := manager.Collect(uint(len(hookNames)), 10*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+
+			callbackMaps := manager.CallbacksPerHookPoint
+			Expect(callbackMaps).Should(HaveKey(hookPointName))
+			Expect(callbackMaps[hookPointName]).Should(HaveLen(len(hookNames)))
+		})
+
+		It("Should find multiple sidecars on differrnt hook points", func() {
```
Comment (booxter) [inline pkg/hooks/manager_test.go]: different

**Comment by rmohr** [comment]
@lxs137 I took the liberty to push the missing formatting changes in an extra commit on your branch. I hope that is fine. Thanks for the contribution :+1: 

/lgtm
/approve

**Comment by lxs137** [comment]
> @lxs137 I took the liberty to push the missing formatting changes in an extra commit on your branch. I hope that is fine. Thanks for the contribution 👍
> 
> /lgtm
> /approve

THX, the Travis CI flow is too confused for me

**Comment by rmohr** [comment]
All tests passing.

**Pull request merged**